lib/si-tls/src/lib.rs

```diff
@@ -33,67 +36,117 @@ pub enum TlsError {
 #[derive(Clone, Debug, Deserialize, Serialize)]
 #[serde(rename_all = "snake_case", tag = "type", content = "source")]
 pub enum CertificateSource {
+    AsString(String),
     Base64(String),
     Path(CanonicalFile),
     Remote(String),
 }
 
-pub struct CertificateResolver;
+impl CertificateSource {
+    /// Load certs from any CertificateSource
+    pub async fn load_certificates(&self) -> Result<Vec<CertificateDer<'static>>> {
+        Self::get_certificate_from_bytes(self.load_certificates_as_bytes().await?).await
+    }
 
-impl CertificateResolver {
-    // Load certificated based on the source location
-    pub async fn load_certificates(
-        source: &CertificateSource,
-    ) -> Result<Vec<CertificateDer<'static>>> {
-        match source {
-            CertificateSource::Path(path) => Self::get_certificate_from_path(path).await,
-            CertificateSource::Base64(data) => {
-                Self::get_certificate_from_base64(data.clone()).await
+    /// Load certs as bytes from any CertificateSource
+    pub async fn load_certificates_as_bytes(&self) -> Result<Vec<u8>> {
+        match self {
+            Self::AsString(string) => Ok(string.as_bytes().to_vec()),
+
+            Self::Base64(data) => Ok(STANDARD.decode(data)?),
+            Self::Path(path) => Ok(std::fs::read(path)?),
```

Comment:
There is also a `tokio::fs::read(path).await?` call that would flip this to async...